iohomework/src/main/java/SalesStatistics.java

```diff
@@ -0,0 +1,88 @@
+import java.util.*;
+
+/**
+ * This class is used to generate statistics based on Clients and Products.
+ * Clients and Products are stored in two separate HashMaps.
+ */
+
+class SalesStatistics {
+
+    // key - product name, value - product object
+    HashMap<String, Product> productSalesHashMap = new HashMap<>();
+
+    // key - client name, value - sales object
+    HashMap<String, Client> clientSalesHashMap = new HashMap<>();
+
+    /**
+     * @param sales represents a Sales Object that is added to the statistics.
+     *              updates both HashMaps.
+     */
+
+    void addSales (Sales sales){
+        Product productName = getProductfromMap(sales.getName());
+        Client clientName = getClientfromMap(sales.getClient());
+        productName.addSoldUnits(sales.getSoldunits());
+        productName.addPrice(sales.getPrice()*sales.getSoldunits());
+        clientName.addUnitsBought(sales.getSoldunits());
+        clientName.addAmountSpent(sales.getPrice()*sales.getSoldunits());
+    }
+
+    /**
+     * @param productName updates the Product Object in HashMap or creates a new one.
+     * @return
+     */
+    private Product getProductfromMap(String productName) {
+        Product product;
+        if (!productSalesHashMap.containsKey(productName)) {
+            product = new Product(productName);
+            productSalesHashMap.put(productName, product);
+        } else {
+            product = productSalesHashMap.get(productName);
+        }
+        return product;
+    }
+
+    /**
+     * @param clientName updates the Client Object in HashMap or creates a new one.
+     * @return
+     */
+    private Client getClientfromMap(String clientName) {
+        Client client;
+        if (!clientSalesHashMap.containsKey(clientName)) {
+            client = new Client(clientName);
+            clientSalesHashMap.put(clientName, client);
+        } else {
+            client = clientSalesHashMap.get(clientName);
+        }
+        return client;
+    }
+
+    List<Product> statsUnitsSold(HashMap<String, Product> productHashMap) {
```

Comment:
Why is that param needed?!

Reply:
I dont understand the question. I have 4 comparisons to make...two on products (by units and by ammount generated) and two on clients (by units bought and by income spent) so I created a list for each.